src/lib/store.ts

```diff
@@ -0,0 +1,78 @@
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
+import { join } from 'path';
+
+import { app, remote } from 'electron';
+
+/** App-specific directory to store information in. */
+const BASE_PATH = join((app || remote.app).getPath('userData'), 'sentry');
+
+/**
+ * Lazily serializes data to a JSON file to persist it beyond application
+ * crashes. When created, it loads data from that file if it already exists.
+ */
+export default class Store<T> {
+  /** Current state of the data. */
+  private data: T;
+  /** Path the JSON file storing the information. */
+  private path: string;
+  /** State whether a flush to disk has been requested in this cycle. */
+  private flushing: boolean = false;
+
+  /**
+   * Creates a new store.
+   *
+   * @param filename A unique filename to store this data.
+   * @param initial An initial value to initialize data with.
+   */
+  constructor(filename: string, private initial?: T) {
+    if (!existsSync(BASE_PATH)) {
+      mkdirSync(BASE_PATH);
```

Comment:
This causes an exception if its the first time the app has been run as the base `AppData` directory does not exist.

Reply:
`AppData` is not created until `makeSingleInstance` is called or you create `BrowserWindows` and a few other assorted APIs.

Reply:
Would this be solved when we lazy create the folder?

Reply:
Somebody could write an Electron app where they don't hit any points that create the `AppData` directory. One example I can think of is an app which starts in the tray and doesn't create any windows immediately. It's safe to check if `AppData` exists and create it if not.

It'll be worth noting in the docs that if you change the location of `AppData`, this needs to be done before starting Sentry. On Windows it common to change it to `%LOCALAPPDATA%\AppName` so you don't fill up roaming profiles.